docs/noise.ipynb

```diff
@@ -62,196 +62,673 @@
+    "An minimal example for defining the channel\n",
     "\n",
     "$$\n",
-    "\\begin{aligned}\n",
-    "\\rho \\rightarrow& \\sum_{k=0}^3 M_k \\rho M_k \\newline\n",
-    "M_0 =& \\sqrt{p} \\begin{bmatrix} 1 & 0  \\cr 0 & \\sqrt{1 - \\gamma} \\end{bmatrix} \\newline\n",
-    "M_1 =& \\sqrt{p} \\begin{bmatrix} 0 & \\sqrt{\\gamma} \\cr 0 & 0 \\end{bmatrix} \\newline\n",
-    "M_2 =& \\sqrt{1-p} \\begin{bmatrix} \\sqrt{1-\\gamma} & 0 \\cr 0 & 1 \\\\ \\end{bmatrix} \\newline\n",
-    "M_3 =& \\sqrt{1-p} \\begin{bmatrix} 0 & 0 \\cr \\sqrt{\\gamma} & 0 \\end{bmatrix}\n",
-    "\\end{aligned}\n",
+    "\\rho \\mapsto (1 - p) \\rho + p Y \\rho Y\n",
     "$$\n",
     "\n",
-    "Where &#947; is the probability of the interaction being dissipative, and\n",
-    "`p` is the probability that the qubit and environment exchange energy. The amplitude damping channel corresponds to `p=1`.\n",
+    "is shown below."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "55240879394e"
+   },
+   "outputs": [],
+   "source": [
+    "\"\"\"Minimal example of defining a custom channel.\"\"\"\n",
+    "class BitAndPhaseFlipChannel(cirq.SingleQubitGate):\n",
+    "    def __init__(self, p: float) -> None:\n",
+    "        self._p = p\n",
+    "    \n",
+    "    def _mixture_(self):\n",
+    "        ps = [1.0 - self._p, self._p]\n",
+    "        ops = [cirq.unitary(cirq.I), cirq.unitary(cirq.Y)]\n",
+    "        return tuple(zip(ps, ops))\n",
+    "    \n",
+    "    def _has_mixture_(self) -> bool:\n",
+    "        return True\n",
+    "    \n",
+    "    def _circuit_diagram_info_(self, args) -> str:\n",
+    "        return f\"BitAndPhaseFlip({self._p})\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "8ffff44ff2b2"
+   },
+   "source": [
+    "> *Note*: The `_has_mixture_` magic method is not strictly required but is recommended."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "2dea6bd62cfe"
+   },
+   "source": [
+    "We can now instantiate this channel and get its mixture:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "0350d9e193ca"
+   },
+   "outputs": [],
+   "source": [
+    "\"\"\"Custom channels can be used like any other channels.\"\"\"\n",
+    "bit_phase_flip = BitAndPhaseFlipChannel(p=0.05)\n",
     "\n",
-    "Cirq provides the helpers `cirq.generalized_amplitude_damp` and\n",
-    "`cirq.amplitude_damp` to construct these noisy gates."
+    "for prob, kraus in cirq.mixture(bit_phase_flip):\n",
+    "    print(f\"With probability {prob}, apply\\n\", kraus, end=\"\\n\\n\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "72486a155131"
+   },
+   "source": [
+    "> *Note:* Since `_mixture_` is defined, we can also use the `cirq.channel` protocol."
```

Comment:
```suggestion
    "Note: Since `_mixture_` is defined, we can also use the `cirq.channel` protocol."
```

"Note:" is a style element used as a callout box on the webpage